src/marked.js

```diff
@@ -164,6 +154,50 @@ marked.use = function(extension) {
   marked.setOptions(opts);
 };
 
+/**
+ * Iterate over every token
+ */
+
+marked.iterateTokens = function(tokens, callback) {
```

Comment:
I'm not sure what the best name for this function should be. A few names that I thought would be good:

- iterateTokens
- mapTokens (this might be confusing because we don't update the tokens based on the return value)
- updateTokens

/cc @styfle @joshbruce @davisjam

Reply:
iterateTokens seems fine

Reply:
I vote no for `mapTokens` because the return type is not an array of tokens.

I think `iterateTokens` is okay but maybe something like `walkTokens` would be better because it implies walking the tree recursively. https://en.wikipedia.org/wiki/Tree_traversal

Reply:
Ah, I like `walkTokens`.